### PR TITLE
[JENKINS-20998] Added timestamp support to DirectoryBrowserSupport.java ...

### PR DESCRIPTION
**Pull request by cjbush**
Added feature requested in JENKINS-20998

### CODE REVIEW

File: core/src/main/java/hudson/model/DirectoryBrowserSupport.java [hunk 1]
```diff
@@ -385,15 +385,21 @@ private static void zip(OutputStream outputStream, VirtualFile dir, String glob)
...
+         */
+        private final String date;
+        
+        /**
          * If the current user can read the file.
          */
         private final boolean isReadable;
 
-        public Path(String href, String title, boolean isFolder, long size, boolean isReadable) {
+        public Path(String href, String title, boolean isFolder, long size, String date, boolean isReadable) {
```
Comment (cjbush) [inline core/src/main/java/hudson/model/DirectoryBrowserSupport.java]: Fixed this

File: core/src/main/java/jenkins/util/VirtualFile.java [hunk 2]
```diff
@@ -339,6 +355,13 @@ public static VirtualFile forFilePath(final FilePath f) {
                     throw (IOException) new IOException(x.toString()).initCause(x);
                 }
             }
+            @Override public String date() throws IOException {
+                try{
+                    return new SimpleDateFormat(VirtualFile.DateFormat).format(f.lastModified());
```
Comment (cjbush) [inline core/src/main/java/jenkins/util/VirtualFile.java]: Fixed this as well

File: core/src/main/java/jenkins/util/VirtualFile.java [hunk 3]
```diff
@@ -134,6 +136,17 @@
...
 
     /**
+     * Gets the file date.
+     * @return a date, or null if inapplicable
+     * @throws IOException  if checking the date failed
+     */
+    public abstract String date() throws IOException;
+    
+    
+    public static final String DateFormat = "MM/dd/yyyy h:mm:ss a";
```
Comment (cjbush) [inline core/src/main/java/jenkins/util/VirtualFile.java]: It's localized now

**Comment by daniel-beck** [comment]
Whoops. So much for closing/reopening the PR to trigger a shiny new build :frowning:

**Comment by daniel-beck** [comment]
@cjbush Are you still interested in getting this into Jenkins? If so, could you restore the repository and resubmit a PR? I'd rather not copy your code, I'm not even sure you'd be okay with that, even when I restore your name on the commits.

I realize we dropped the ball for way too long here…

**Comment by cjbush** [comment]
Sure, I'll get around to it. Right now I'm trying to figure out what I broke in the Emma plugin getting it up to the 2.0 API.

**Pull request closed without merging**